gpflow/conditionals/util.py

```diff
@@ -251,23 +306,21 @@ def fully_correlated_conditional(Kmn, Kmm, Knn, f, *, full_cov=False, full_outpu
     :param Kmm: [M, M]
     :param Knn: [N, P] or [N, P, N, P]
     :param f: data matrix, [M, 1]
-    :param q_sqrt: [1, M, M]  or [1, L]
+    :param q_sqrt: [1, M, M] or [1, L]
     :param full_cov: calculate covariance between inputs
     :param full_output_cov: calculate covariance between outputs
     :param white: use whitened representation
     :return:
         - mean: [N, P]
         - variance: [N, P], [N, P, P], [P, N, N], [N, P, N, P]
     """
-    m, v = fully_correlated_conditional_repeat(Kmn,
-                                               Kmm,
-                                               Knn,
-                                               f,
-                                               full_cov=full_cov,
-                                               full_output_cov=full_output_cov,
-                                               q_sqrt=q_sqrt,
-                                               white=white)
-    return m[0, ...], v[0, ...]
+    mean, var = fully_correlated_conditional_repeat(
+        Kmn, Kmm, Knn, f,
+        full_cov=full_cov,
+        full_output_cov=full_output_cov,
+        q_sqrt=q_sqrt,
+        white=white)
+    return tf.squeeze(mean, axis=0), tf.squeeze(var, axis=0)
```

Comment:
is `tf.squeeze` as performant as `Tensor.__getitem__`?

Reply:
`tf.squeeze` checks that the dimension you're squeezing actually only has length 1. I think this extra check is valuable, and I don't expect it to have a significant performance impact relative to any of the matmuls.